execution_engine/src/storage/trie/mod.rs

```diff
@@ -143,23 +136,29 @@ impl<'de> Deserialize<'de> for PointerBlock {
             type Value = PointerBlock;
 
             fn expecting(&self, formatter: &mut Formatter) -> fmt::Result {
-                formatter.write_str("Sparse representation of a PointerBlock.")
+                formatter.write_str("sparse representation of a PointerBlock")
             }
 
-            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+            fn visit_map<M>(self, mut access: M) -> Result<Self::Value, M::Error>
             where
-                A: SeqAccess<'de>,
+                M: MapAccess<'de>,
             {
-                let mut ret = PointerBlock::new();
+                let mut pointer_block = PointerBlock::new();
+
                 // Unpack the sparse representation
-                while let Some(indexed_pointer) = seq.next_element()? {
-                    let (idx, ptr): (u8, Pointer) = indexed_pointer;
-                    ret[idx as usize] = Some(ptr);
+                while let Some((index, pointer_block_value)) =
+                    access.next_entry::<usize, Pointer>()?
+                {
+                    let element = pointer_block.0.get_mut(index).ok_or_else(|| {
+                        de::Error::custom(format!("invalid index {} in pointer block value", index))
+                    })?;
+                    *element = Some(pointer_block_value);
```

Comment:
```suggestion
                    pointer_block[index as usize] = Some(pointer_block_value);
```

Pointer block radix is 256 so a `u8` is never out of bounds.

Reply:
[Here][1] is an older example where we've done this.

[1]: https://github.com/xcthulhu/casper-node/blob/ae40fa08138d2a53286b3f8cfc2c09d28fc0fd2b/execution_engine/src/storage/trie/mod.rs#L123

Reply:
The problem here is that `index` is serialized as a `usize`, not a `u8`.

We could change the serialization to use `u8` rather than `usize` and this direct access would be appropriate, but I was worried that if we change the radix to something other than 256, we'd run the risk of making this unsafe.  A malformed, serialized `PointerBlock` could cause a panic - a simple attack.

If we're convinced that we'll never change the radix from 256, then I'm fine with this suggestion (I'd change the serialization to use `u8` for the index)

Reply:
Makes sense.  We should probably consider changing the old code then, but that seems out of scope.